avaframe/com1DFA/particleInitialisation.py

```diff
@@ -0,0 +1,246 @@
+    particles['ID'] = np.arange(particles['nPart'])
+    particles['nID'] = particles['nPart']
+    particles['parentID'] = np.arange(particles['nPart'])
+    # reset particle properties
+    nPart = particles['nPart']
+    particles['ux'] = np.zeros(nPart)
+    particles['uy'] = np.zeros(nPart)
+    particles['uz'] = np.zeros(nPart)
+    particles['s'] = np.zeros(nPart)
+    particles['l'] = np.zeros(nPart)
+    particles['stoppCriteria'] = False
+    particles['kineticEne'] = 0.0
+    particles['peakKinEne'] = 0.0
+    particles['peakMassFlowing'] = 0.0
+    particles['potentialEne'] = np.sum(cfg['GENERAL'].getfloat('gravAcc') * particles['m'] * particles['z'])
+    # TODO: note particle flow depth is not updated- this is done in updateFieldsC in the next step as the flow
+    # depth is currently computed from the mass and an interpolation on the grid
+
+    # for final configuration get neighbors and update fields
+    particles = DFAfunC.getNeighborsC(particles, dem)
+    particles, fields = DFAfunC.updateFieldsC(cfg['GENERAL'], particles, dem, fields)
+
+    fields['pfv'] = fields['FD']
+    fields['ppr'] = fields['P']
+    fields['pfd'] = fields['FV']
+
+    # save particles to file for visualisation
+    avaDir = pathlib.Path(cfg['GENERAL']['avalancheDir'])
+    outDir = avaDir / 'Outputs' / 'com1DFA' / 'particlesIni'
+    particleTools.savePartToCsv(cfg['VISUALISATION']['particleProperties'], particlesList, outDir)
+    log.info('Initialising particles finalized, total mass: %.2f, number of particles: %d' % (np.sum(particles['m']), particles['nPart']))
+
+    return particles, fields
+
+
+def resetMassPerParticle(cfg, particles, dem, relRaster):
+    """ recompute mass of particles according to their location with respect to relRaster
+
+        Parameters
+        ------------
+        cfg: configparser object
+            configuration settings
+        particles: dict
+            dictionary with particles properties
+        dem: dict
+            dictionary with info on dem
+        relRaster: np.array
+            raster of release thickness values
+
+        Returns
+        ---------
+        particles: dict
+            updated particles dictionary with new mass
+    """
+
+    ncols = dem['header']['ncols']
+    nrows = dem['header']['nrows']
+    indPartInCell = particles['indPartInCell']
+    partInCell = particles['partInCell']
+    rho = cfg['GENERAL'].getfloat('rho')
+
+    indRelY, indRelX = np.nonzero(relRaster)
+    particles['mIni'] = np.zeros(particles['nPart'])
+    for indRelx, indRely in zip(indRelX, indRelY):
+        # compute number of particles for this cell
+        hCell = relRaster[indRely, indRelx]
+        volCell = dem['areaRaster'][indRely, indRelx] * hCell
+        massCell = volCell * rho
+        ic = indRelx + ncols * indRely
+        iStart = indPartInCell[ic]
+        iStop = indPartInCell[ic+1]
+        indParts = partInCell[iStart:iStop]
+        mNew = massCell / len(indParts)
```

Comment:
could we also add the option to do the bilinear interpolation method on flow depth here too?

Reply:
made an issue